G1ANT.Addon.Selenium/Api/SeleniumDriverResourceDescription.cs

```diff
@@ -1,84 +1,118 @@
+        private string resourceName64;
+        private const int killingProcessTimeout = 10000;
+        private const int createFileTimeout = 10000;
 
         public SeleniumDriverResourceDescription(string driverName, string resourceName)
         {
             Init(driverName, resourceName, resourceName);
         }
 
-        public SeleniumDriverResourceDescription(string driverName, string resourceName32, string resourceName64)
-        {
-            Init(driverName, resourceName32, resourceName64);
-        }
+        public SeleniumDriverResourceDescription(string driverName, string resourceName32, string resourceName64) 
+            => Init(driverName, resourceName32, resourceName64);
 
         public void UnpackIfNeeded(Assembly assembly, string unpackFolder)
         {
-            var resourceName = Environment.Is64BitOperatingSystem ? ResourceName64 : ResourceName32;
+            var resourceName = Environment.Is64BitOperatingSystem ? resourceName64 : resourceName32;
             var resourceBinary = assembly.GetResourceBytes(resourceName);
 
             if (resourceBinary == null)
                 throw new ApplicationException($"Driver '{resourceName}' is not embedded in the Addon.Selenium.");
-            if (DriverNeedsUpdate(DriverName, resourceBinary, unpackFolder))
-            {
-                UpdateDriverFromResource(DriverName, resourceBinary, unpackFolder);
-            }
+
+            if (DriverNeedsUpdate(driverName, resourceBinary, unpackFolder))
+                UpdateDriverFromResource(driverName, resourceBinary, unpackFolder);
         }
 
         private void Init(string driverName, string resourceName32, string resourceName64)
         {
-            DriverName = driverName;
-            ResourceName32 = resourceName32;
-            ResourceName64 = resourceName64;
+            this.driverName = driverName;
+            this.resourceName32 = resourceName32;
+            this.resourceName64 = resourceName64;
         }
 
-        private bool DriverNeedsUpdate(string filename, byte[] resourceData, string unpackFolder)
+        private bool DriverNeedsUpdate(string filename, byte[] resourceData, string unpackFolder) 
+            => !DoesFileExist(unpackFolder, filename) || !AreFilesOfTheSameLength(resourceData.Length, unpackFolder, filename);
+
+        private void UpdateDriverFromResource(string filename, byte[] resourceData, string unpackFolder)
         {
-            return !DoesFileExist(unpackFolder, filename) || !AreFilesOfTheSameLength(resourceData.Length, unpackFolder, filename);
+            KillWorkingProcess(Path.GetFileNameWithoutExtension(filename), unpackFolder);
+            var startTickCount = Environment.TickCount;
+            while (!CreateDriverFileFromResoure(filename, resourceData, unpackFolder)
+                && !TimeoutOccured(startTickCount, createFileTimeout))
+                ;
+
+            if (TimeoutOccured(startTickCount, createFileTimeout))
```

Comment:
Exception can be thrown even when CreateDriverFileFromResoure method succeed because you perform another timeout check after CreateDriverFileFromResoure method